src/metabase/models/data_permissions.clj

```diff
@@ -144,6 +145,63 @@
       (or (coalesce perm-type perm-values)
           (least-permissive-value perm-type)))))
 
+(mu/defn data-permissions-graph-for-user
```

Comment:
Just curious: Why not let `data-permissions-graph` take a set of db-ids or graph-ids, and then use user-id -> permissionGroups, and then use that set to calculate this?